ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+        stderr      => \$err,
+        keeps_state => $keeps_state );
+
+    $opts{stdin} = $stdin if defined($stdin);
+
+    my $cmd = CAF::Process->new( $command, %opts );
+
+    $cmd->info("$why");
+    $cmd->execute();
+    $yum_error = $err;
+    if ( $err && ( !defined($error_ok) || $error_ok != 1 ) ) {
+        $self->warn("$why produced warnings: $err") if ( $err && ( !defined($error_ok) || $error_ok != 1 ) );
+        $yum_warned = 1;
+    }
+    $self->verbose("$why output:\n$out") if ( defined($out) );
+    if ( $? ||
+        ( $err && $err =~ m{^(?:Error|Failed|
+                      (?:Could \s+ not \s+ match)|
+                      (?:Transaction \s+.*rror)|
+                      (?:Unknown \s+ group \s+  package \s+ type) |
+                      (?:.*requested \s+ URL \s+ returned \s+ error))}oxmi ) ||
+        ( $out && ( @missing = ( $out =~ m{^No package (.*) available}omg ) ) )
+      ) {
+        return undef if ( !defined($error_ok) || $error_ok != 1 );
+        $self->$error_logger( "Failed $why: ", $err || "(empty/undef stderr)" );
+        if (@missing) {
+            $self->$error_logger( "Missing packages: ", join ( " ", @missing ) );
+        }
+        return $err;
+    }
+    return $cmd->{NoAction} || $out;
+}
+
+sub execute_yum_command_with_recovery
+{
+    my ( $self, $command, $why, $keeps_state, $stdin, $error_logger, $error_ok ) = @_;
+    my $status = $self->execute_yum_command($command, $why, $keeps_state, $stdin, $error_logger, $error_ok);
+    my $packages_modified = 0;
+
+    if ( !defined($status) ) {
+	my @lines = split /\n/, $yum_error;
+	foreach my $line (@lines) {
+	    my $cf = index($line, 'conflicts with file from package');
+	    if ( $cf != -1) {
+		my $pkg = $line;
+		$pkg =~ s,^.*package ,,;
+		$pkg =~ s,-[0-9]*:,-,;
+		$self->execute_yum_command(["rpm -e --nodeps $pkg"], "removing $pkg to avoid file conflict");
+		$packages_modified = 1;
+	    }
+	}
+	if ( $packages_modified ) {
+	    $status = $self->execute_yum_command($command, 'retrying: '.$why, $keeps_state, $stdin, $error_logger, $error_ok);
+	}
+    }
+    return $status;
+}
+
+sub Configure
+{
+    my ( $self, $config ) = @_;
+
+    # Make sure stdout and stderr is flushed every time to not to
+    # put mess on serial console
+    autoflush STDOUT 1;
+    autoflush STDERR 1;
+
+    # We are parsing some outputs in this component. We must set a
+    # locale that we can understand.
+    local $ENV{LANG}   = 'C';
+    local $ENV{LC_ALL} = 'C';
+
+    my $repos = $config->getElement(REPOS_TREE)->getTree();
+    my $t     = $config->getElement(CMP_TREE)->getTree();
+    
+    # Display system info
+    if ( defined($t->{quattor_os_release}) ) {
+	$self->info("target OS build: ", $t->{quattor_os_release});
+    }
+    
+    my $spmaver = $self->execute_yum_command( ["rpm -q --qf %{NAME}-%{VERSION}-%{RELEASE}.%{ARCH} ncm-spma"], "checking for spma version", 1);
+    $self->info("SPMA version: ", $spmaver);
```

Comment:
any particular reason? would https://github.com/quattor/maven-tools/pull/62 help instead? (all components version should better be logged by ncm-ncd. https://github.com/quattor/ncm-ncd/issues/48)